model/task_queue_service.go

```diff
@@ -60,16 +60,48 @@ func (s *taskDispatchService) FindNextTask(distroID string, spec TaskSpec) (*Tas
 }
 
 func (s *taskDispatchService) RefreshFindNextTask(distroID string, spec TaskSpec) (*TaskQueueItem, error) {
+	start := time.Now()
 	distroDispatchService, err := s.ensureQueue(distroID)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
+	grip.DebugWhen(time.Since(start).Seconds() > 1, message.Fields{
+		"step":            "RefreshFindNextTask",
+		"distro":          distroID,
+		"dispatcher_type": distroDispatchService.Type(),
+		"op":              "ensureQueue",
+		"duration":        time.Since(start),
+		"duration_secs":   time.Since(start).Seconds(),
+	})
+	start = time.Now()
 
 	if err := distroDispatchService.Refresh(); err != nil {
 		return nil, errors.WithStack(err)
 	}
-
-	return distroDispatchService.FindNextTask(spec), nil
+	grip.DebugWhen(time.Since(start).Seconds() > 1, message.Fields{
+		"step":            "RefreshFindNextTask",
+		"distro":          distroID,
+		"dispatcher_type": distroDispatchService.Type(),
+		"op":              "Refresh",
+		"duration":        time.Since(start),
+		"duration_secs":   time.Since(start).Seconds(),
+	})
+	start = time.Now()
+
+	item := distroDispatchService.FindNextTask(spec)
+	msg := message.Fields{
+		"step":            "RefreshFindNextTask",
+		"distro":          distroID,
+		"dispatcher_type": distroDispatchService.Type(),
+		"op":              "FindNextTask",
+		"duration":        time.Since(start),
+		"duration_secs":   time.Since(start).Seconds(),
+	}
+	if item != nil {
+		msg["item"] = item.Id
+	}
+	grip.DebugWhen(time.Since(start).Seconds() > 1, msg)
```

Comment:
start keeps getting reset so I don't think this does what you want it to

Reply:
It finds steps that take longer than a second, which I think is the right thing.